Framework/Crystal/src/SCDCalibratePanels.cpp

```diff
@@ -1967,50 +1020,52 @@ void SCDCalibratePanels::saveXmlFile(
   ParameterMap_sptr pmap = instrument->getParameterMap();
 
   // write out the detector banks
-  for (const auto &Group : Groups) {
-    for (const auto &bankName : Group) {
-      oss3 << "<component-link name=\"" << bankName << "\">\n";
-
-      boost::shared_ptr<const IComponent> bank =
-          instrument->getComponentByName(bankName);
-
-      Quat RelRot = bank->getRelativeRot();
-
-      double rotx, roty, rotz;
-
-      SCDCalibratePanels::Quat2RotxRotyRotz(RelRot, rotx, roty, rotz);
-      writeXmlParameter(oss3, "rotx", rotx);
-      writeXmlParameter(oss3, "roty", roty);
-      writeXmlParameter(oss3, "rotz", rotz);
-
-      V3D pos1 = bank->getRelativePos();
-      writeXmlParameter(oss3, "x", pos1.X());
-      writeXmlParameter(oss3, "y", pos1.Y());
-      writeXmlParameter(oss3, "z", pos1.Z());
-
-      vector<double> oldScalex =
-          pmap->getDouble(bank->getName(), string("scalex"));
-      vector<double> oldScaley =
-          pmap->getDouble(bank->getName(), string("scaley"));
-
-      double scalex, scaley;
-      if (!oldScalex.empty())
-        scalex = oldScalex[0];
-      else
-        scalex = 1.;
-
-      if (!oldScaley.empty())
-        scaley = oldScaley[0];
-      else
-        scaley = 1.;
-
-      oss3 << "  <parameter name =\"scalex\"><value val=\"" << scalex
-           << "\" /> </parameter>\n";
-      oss3 << "  <parameter name =\"scaley\"><value val=\"" << scaley
-           << "\" /> </parameter>\n";
-      oss3 << "</component-link>\n";
-    } // for each bank in the group
-  }   // for each group
+  for (int i = 0; i < static_cast<int>(AllBankNames.size()); ++i) {
+    std::set<string>::iterator it = AllBankNames.begin();
+    advance(it, i);
+    std::string bankName = *it;
```

Comment:
Since the loop variable `i` isn't required and calling `advance` on std::set is slow, a range-based for loop over AllBankNames should be significantly faster.